### PR TITLE
Updated dependencies to fix vulnerabilities

### PR DESCRIPTION
**Pull request by hjobrien**
mostly non breaking updates, but had to switch to gulp 4.x which was breaking. I think my changes preserve functionality, but I had to ignore watching `script.js` for the following reason:

On the old (gulp 3.x) version, we had a task which was run whenever a js file was modified.
As best I can tell, this involved concatenating content to a `script.js` file

When I updated to gulp 4.x, I noticed that it appeared as though there were changes being made constantly to the js files (even though I hadn't made any changes)

After some looking around, it seems like we were watching for changes to `.js` files, 
then calling a task which itself changed a `.js` file, which caused the listener to fire again, over and over.

I reverted back to gulp3.x, and for some reason this was not happening there, I'm not sure why.

I fixed this by ignoring live changes to `script.js`, but because this was working in the old gulp version without the ignore, I imagine that there is a better way, though I'm not sure of what it is.

### CODE REVIEW

**Comment by hjobrien** [comment]
tagging issue #23

**Comment by hjobrien** [comment]
I can gitignore `package-lock.json` but I wanted to avoid modifying unrelated files for this PR

File: gulpfile.js [hunk 1]
```diff
@@ -30,17 +30,17 @@ gulp.task('js', function() {
...
 
-    gulp.watch('sass/*.scss', ['sass']);
+    gulp.watch('sass/*.scss', gulp.parallel(['sass']));
     gulp.watch('*.html').on('change', reload);
     gulp.watch('**/*.html').on('change', reload);
     // gulp.watch('**/*.js', ['js']).on('change', reload);
-    gulp.watch('*.js', ['js']).on('change', reload);
-    gulp.watch('js/**/*.js', ['js']).on('change', reload);
-});
+    gulp.watch(['*.js', '!js/script.js'], gulp.parallel(['js'])).on('change', reload);
```
Comment (wuisawesome) [inline gulpfile.js]: This looks suspect. Compare to the watching section of this: "https://www.joezimjs.com/javascript/complete-guide-upgrading-gulp-4/" 

https://www.joezimjs.com/javascript/complete-guide-upgrading-gulp-4/

**Comment by wuisawesome** [comment]
> I can gitignore `package-lock.json` but I wanted to avoid modifying unrelated files for this PR
> But yes, it is autogenerated

`git checkout origin/master package-lock.json`?

**Pull request merged**